### PR TITLE
MRG: Allow starting CoregistrationUI in fullscreen mode

### LINKED ISSUE #10745: Add config variable to control whether to start coregistration UI in fullscreen mode
_Originally posted by @larsoner in https://github.com/mne-tools/mne-python/issues/10284#issuecomment-1027217610_

[...] we have the config vars `"MNE_COREG_SCENE_WIDTH": "845",` and corresponding HEIGHT that should in principle be read on init, used, and then written on close (not sure all these steps have been implemented yet). We should just add a `FULLSCREEN` that is true or false (default) that gets respected, too. Want to implement it here?

### PR DESCRIPTION
**Pull request by hoechenberger**
When doing coregistration, I found my very first interaction with the GUI was always a click to maximize the window.

This PR adds a parameter to `mne.gui.coregistration()` to allow starting up in fullscreen mode right away.

I wonder if this could even become the default one day.

Closes #10745

### CODE REVIEW

File: mne/gui/_coreg.py [hunk 1]
```diff
@@ -81,7 +81,11 @@ class CoregistrationUI(HasTraits):
     show : bool
         Display the window as soon as it is ready. Defaults to True.
     block : bool
-        If True, start the Qt application event loop. Default to False.
+        Whether to halt program script execution until the GUI has been closed
```
Comment (hoechenberger) [inline mne/gui/_coreg.py:84]:
```suggestion
        Whether to halt program execution until the GUI has been closed
```

**Comment by larsoner** [comment]
> I wonder if this could even become the default one day.

-1 on the default, +1 on using it if the user does, though. In other words, we have the config vars `"MNE_COREG_SCENE_WIDTH": "845",` and corresponding HEIGHT that should in principle be read on init, used, and then written on close (not sure all these steps have been implemented yet). We should just add a `FULLSCREEN` that is true or false (default) that gets respected, too. Want to implement it here?

**Comment by hoechenberger** [comment]
> We should just add a `FULLSCREEN` that is true or false (default) that gets respected, too. Want to implement it here?

I can give it a shot!

Although just today during a debugging session with @GuillaumeFavelier we started discussing whether the MNE config is actually more harm- than useful in some cases (turned out Guillaume had some defaults auto-applied from the config that he wasn't aware of)

**Comment by larsoner** [comment]
> personally I'm not a fan of the config and the "silent" modifications it does to MNE behavior, and prefer explicit kwargs instead

It used to be the case that some people preferred things like "mark inside" and others didn't in their workflows, so it made sense (to me at least) to have these be preserved at each user's end, with a way to switch them on and off in the GUI easily. We've been eliminating at least some of these from the GUI so maybe they aren't necessary anymore...

I think things like window size are definitely useful and should be kept. Same thing with fullscreen

**Comment by hoechenberger** [comment]
Ok I'll try to get this to work tomorrow 👍

**Comment by larsoner** [comment]
I don't see this as a blocker for 1.0 so I'll move the milestone

**Comment by hoechenberger** [comment]
@larsoner @agramfort 
I was hoping to get this into 1.1 as well

It works nicely for me locally, let's wait for CI results

@larsoner I've dropped the idea of adding a config setting for now, and I'm defaulting to `fullscreen=False`. I think we can always add a config variable later and transition to `fullscreen=None`. Is that okay with you?

I just wanted to finally get this functionality merged.

File: mne/utils/docs.py [hunk 2]
```diff
@@ -1301,6 +1301,11 @@ def _reflow_param_docstring(docstring, has_first_line=True, width=75):
     .. versionadded:: 0.21
 """
 
+docdict['fullscreen'] = """
+fullscreen : bool
+    Whether to start in fullscreen or windowed mode (``False``, default).
+"""
```
Comment (hoechenberger) [inline mne/utils/docs.py:1307]:
```suggestion
fullscreen : bool
    Whether to start in fullscreen (``True``) or windowed mode
    (``False``, default).
"""
```

**Comment by larsoner** [comment]
GitHub seems to be having a bad day today. I'll try restarting CIs later

FYI I pushed a tiny commit to make config val support work @hoechenberger

**Comment by hoechenberger** [comment]
Oh, cool, nice addition!

**Comment by larsoner** [comment]
I am at a loss for the GH download problem so I opened https://github.com/orgs/github-community/discussions/18543

**Pull request merged**